File: splits.js
const words="asdff\nastgrw3h\r\nwtegole\rkserlhge3t\nearsgh\nergh\rsagr\r\nerghe\r";
var wordList;

console.log("JavaScript:string.split(Regex)");
console.time("sw");
for(let i=0;i<1000000;i=0|i+1){
	wordList=words.split(/\r\n|\n|\r/);
}
console.timeEnd("sw");
console.log(wordList.join(","));

console.log("JavaScript:string.replace & split(string)");
console.time("sw");
for(let i=0;i<1000000;i=0|i+1){
	wordList=words.replace(/\r\n/g,"\n").replace(/\r/g,"\n").split("\n");
}
console.timeEnd("sw");
console.log(wordList.join(","));

console.log("JavaScript:string.replace & split(string)");
console.time("sw");
for(let i=0;i<1000000;i=0|i+1){
	wordList=words.split("\r\n").join("\n").split("\r").join("\n").split("\n");
}
console.timeEnd("sw");
console.log(wordList.join(","));

File: splits.py
from time import time
import re

words="asdff\nastgrw3h\r\nwtegole\rkserlhge3t\nearsgh\nergh\rsagr\r\nerghe\r"

print("Python3:string.splitlines")
sw=time();
for i in range(1000000):
	wordList=words.splitlines()
print((time()-sw)*1000)
print(",".join(wordList))

print("Python3:string.replace & split")
sw=time();
for i in range(1000000):
	wordList=words.replace("\r\n","\n").replace("\r","\n").split("\n")
print((time()-sw)*1000)
print(",".join(wordList))

print("Python3:re.split")
sw=time();
for i in range(1000000):
	wordList=re.split("\r\n|\n|\r",words)
print((time()-sw)*1000)
print(",".join(wordList))
